ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -140,9 +320,21 @@ public int compareTo(StoreKey o) {
 
     int result = version.compareTo(other.version);
     if (result == 0) {
```

Comment:
Sure, i just feel a bit hesitate to treat two blobs the same if all fields are the same but flag different. Would rather prefer this more rigorous way but of cause need to ensure that this is compatible for v1 blobs.